components/eam/src/chemistry/mozart/mo_chm_diags.F90

```diff
@@ -1537,41 +1558,55 @@ subroutine gaschmmass_diags( lchnk, ncol, vmr, vmr_old, pdeldry, mbar, rdelt, fl
+                        do k = gaschmbudget_2D_L1_s, gaschmbudget_2D_L1_e
+                          wrk_sum(:ncol) = wrk_sum(:ncol) + wrk(:ncol,k)
+                        enddo
+                        call outfld( trim(solsym(m))//'_'//flag(1:5)//'_L1', wrk_sum(:ncol), ncol ,lchnk )
           
-               wrk_sum(:ncol) = 0.0_r8
-               do k = gaschmbudget_2D_L2_s, gaschmbudget_2D_L2_e
-                  wrk_sum(:ncol) = wrk_sum(:ncol) + wrk(:ncol,k)
-               enddo
-               call outfld( trim(solsym(m))//'_'//flag(1:5)//'_L2', wrk_sum(:ncol), ncol ,lchnk )
+                        wrk_sum(:ncol) = 0.0_r8
+                        do k = gaschmbudget_2D_L2_s, gaschmbudget_2D_L2_e
+                           wrk_sum(:ncol) = wrk_sum(:ncol) + wrk(:ncol,k)
+                        enddo
+                        call outfld( trim(solsym(m))//'_'//flag(1:5)//'_L2', wrk_sum(:ncol), ncol ,lchnk )
          
-               wrk_sum(:ncol) = 0.0_r8
-               do k = gaschmbudget_2D_L3_s, gaschmbudget_2D_L3_e
-                  wrk_sum(:ncol) = wrk_sum(:ncol) + wrk(:ncol,k)
-               enddo
-               call outfld( trim(solsym(m))//'_'//flag(1:5)//'_L3', wrk_sum(:ncol), ncol ,lchnk )
+                        wrk_sum(:ncol) = 0.0_r8
+                        do k = gaschmbudget_2D_L3_s, gaschmbudget_2D_L3_e
+                           wrk_sum(:ncol) = wrk_sum(:ncol) + wrk(:ncol,k)
+                        enddo
+                        call outfld( trim(solsym(m))//'_'//flag(1:5)//'_L3', wrk_sum(:ncol), ncol ,lchnk )
         
-               wrk_sum(:ncol) = 0.0_r8
-               do k = gaschmbudget_2D_L4_s, gaschmbudget_2D_L4_e
-                  wrk_sum(:ncol) = wrk_sum(:ncol) + wrk(:ncol,k)
-               enddo
-               call outfld( trim(solsym(m))//'_'//flag(1:5)//'_L4', wrk_sum(:ncol), ncol ,lchnk )
-
-            elseif (len(flag) >= 6 .and. flag(6:10) == '_trop') then
-               wrk_sum(:ncol) = 0.0_r8
-               if (trim(solsym(m))=='O3' .or. trim(solsym(m))=='O3LNZ' .or. &
-                   trim(solsym(m))=='N2OLNZ' .or. trim(solsym(m))=='CH4LNZ') then
-                  do k = 1, pver
-                     wrk_sum(:ncol) = wrk_sum(:ncol) + wrk(:ncol,k) * tropFlagInt(:ncol,k)
-                  enddo
-                  call outfld( trim(solsym(m))//'_'//flag, wrk_sum(:ncol), ncol ,lchnk )
-               endif
+                       wrk_sum(:ncol) = 0.0_r8
+                       do k = gaschmbudget_2D_L4_s, gaschmbudget_2D_L4_e
+                          wrk_sum(:ncol) = wrk_sum(:ncol) + wrk(:ncol,k)
+                       enddo
+                       call outfld( trim(solsym(m))//'_'//flag(1:5)//'_L4', wrk_sum(:ncol), ncol ,lchnk )
+                   endif !kzm ++
+            !kzm ++       
+            !elseif (len(flag) >= 6 .and. flag(6:10) == '_trop') then
+            elseif (len(flag) >= 6) then 
+                    if (flag(6:10) == '_trop') then
```

Comment:
Similar as above, the situation of len(flag) < 10 is not accounted for, which would fail substring referencing, unless such case never exist.

Reply:
@wlin7 Yes, len(flag) >=10 would be better here